### ISSUE TITLE
issue about the cosine similarity loss function

### ISSUE THREAD

**Issue by sonnefred** [facebookresearch/simsiam#14]
Hi,
I'm using simsiam to do semantic segmentation task, and with the cosine similarity loss function I can get some negative values during the training process. How should I optimize such loss function? Thanks!

**Comment by endernewton** [comment]
With cosine similarity, the loss can indeed go below 0 (-1 is the lower bound iirc). I think one can directly use any optimizer to minimize the loss, as the range of the loss theoretically does not affect computing gradient and doing gradient based updates.
